test/e2e-tests.sh

```diff
@@ -16,29 +16,106 @@
 
+TMPF=$(mktemp /tmp/.mm.XXXXXX)
+clean() { rm -f ${TMPF}; }
+trap clean EXIT
+
+
 # Setup a test cluster.
-[[ -z ${LOCAL_CI_RUN} ]] && initialize $@
+[[ -z ${LOCAL_CI_RUN} ]] && {
+    # Initialize cluster
+    initialize $@
+
+    # Install the latest Tekton CRDs.
+    kubectl apply --filename https://storage.googleapis.com/tekton-releases/latest/release.yaml
+}
+
+# You can ignore some yaml tests by providing the TEST_YAML_IGNORES variable
+# with the test name separated by a space, for example:
+#
+# TEST_YAML_IGNORES="kaniko s2i"
+#
+# will ignore the kaniko and s2i tests,
+#
+TEST_YAML_IGNORES=${TEST_YAML_IGNORES:-""}
 
-# Install the latest Tekton CRDs.
-kubectl apply --filename https://storage.googleapis.com/tekton-releases/latest/release.yaml
+# Allow ignoring some yaml tests, space separated, should be the basename of the test
+#
+# Disabled by default:
+#
+# s2i: task is not working by default on vanilla kubernetes (ie not openshift) so disabling it here
+TEST_TASKRUN_IGNORES=${TEST_TASKRUN_IGNORES:-"s2i"}
 
-# Allow ignoring some tests, space separated, should be the basename of the test
-# i.e: kn-deployer
-IGNORES=${IGNORES:-}
+# Specify the default serviceAccount
+SERVICE_ACCOUNT=${SERVICE_ACCOUNT:-default}
 
 set -ex
 set -o pipefail
-# Validate that all the Task CRDs in this repo are valid by creating them in a NS.
-readonly ns="task-ns"
-kubectl create ns "${ns}" || true
-for f in $(find ${REPO_ROOT_DIR} -maxdepth 2 -name '*.yaml'); do
-    skipit=
-    for ignore in ${IGNORES};do
-        [[ ${ignore} == $(basename $(echo ${f%.yaml})) ]] && skipit=True
+
+function test_yaml_can_install() {
+    # Validate that all the Task CRDs in this repo are valid by creating them in a NS.
+    readonly ns="task-ns"
+    kubectl create ns "${ns}" || true
+    for runtest in $(find ${REPO_ROOT_DIR} -maxdepth 2 -name '*.yaml'); do
+        skipit=
+        for ignore in ${TEST_YAML_IGNORES};do
+            [[ ${ignore} == $(basename $(echo ${runtest%.yaml})) ]] && skipit=True
+        done
+        [[ -n ${skipit} ]] && break
+        echo "Checking ${runtest}"
+        kubectl -n ${ns} apply -f <(sed "s/namespace:.*/namespace: task-ns/" "${runtest}")
     done
-    [[ -n ${skipit} ]] && break
-    echo "Checking ${f}"
-    kubectl -n ${ns} apply -f <(sed "s/namespace:.*/namespace: task-ns/" "${f}")
-done
+}
+
+function test_task_creation() {
+    for runtest in */tests;do
+        testname=${runtest%%/*}
+        tns="${testname}-$$"
+        skipit=
+
+        for ignore in ${TEST_TASKRUN_IGNORES};do
+            [[ ${ignore} == ${testname} ]] && skipit=True
+        done
+        [[ -n ${skipit} ]] && continue
+
+        kubectl create namespace ${tns}
+
+        # Install the task itself first
+        for yaml in ${testname}/*.yaml;do
+            sed "s/%SERVICE_ACCOUNT%/${SERVICE_ACCOUNT}/g" < ${yaml} > ${TMPF}
+            [[ -f ${testname}/tests/pre-apply-task-hook.sh ]] && source ${testname}/tests/apply-task-hook.sh
```

Comment:
Looks like it checks for file `pre-apply-task-hook.sh` but sources file `apply-task-hook.sh` (missing `pre-` prefix).